### PR TITLE
ams: distribute sysmodules in single file as stratosphere.romfs

### LINKED ISSUE #927: Multiple "contents" folders: one for system titles, one for user-provided titles
What would you think about having two `contents` folders:
- one containing the "system" titles currently bundled in the releases
- another containing the user-provided titles only (would be empty or left out in release archives)

That would:
- allow knowing which titles are user installed versus which are Atmos reimplementations
- prevent accidental deletion of "system" titles
- make it easy to reset the Atmos install (just empty out the user provided titles folder)

The use case here is someone installed too many sysmodules (preventing the console from booting) and wanted to remove all of them. In the end they had to remove the contents folder and put back the files from the Atmos release.

### PR DESCRIPTION
**Pull request by SciresM**
After this PR is merged, atmosphere will no longer use the /contents/ folder for its system modules.

This PR will include a mechanism for cleaning up pre-ams-0.19.0 contents, which can be removed further in the future.

This closes #927, and is critical to ever tackling #704.

### CODE REVIEW

File: libraries/libstratosphere/source/fs/fs_code.cpp [hunk 1]
```diff
@@ -20,6 +20,55 @@ namespace ams::fs {
...
+                auto sd_guard = SCOPE_GUARD { fs::Unmount("#strat-romfs-sd"); };
+
+                /* Open sd:/atmosphere/stratosphere.romfs. */
+                fs::FileHandle stratosphere_romfs_file;
+                R_TRY(fs::OpenFile(std::addressof(stratosphere_romfs_file), "#strat-romfs-sd:/atmosphere/stratosphere.romfs", fs::OpenMode_Read));
+
+                /* Setup the storage. */
+                /* NOTE: This owns the file, and so on failure it will be closed appropriately. */
+                std::construct_at(GetPointer(g_stratosphere_romfs_storage), stratosphere_romfs_file, true);
+                auto storage_guard = SCOPE_GUARD { std::destroy_at(GetPointer(g_stratosphere_romfs_storage)); };
```
Comment (misson20000) [inline libraries/libstratosphere/source/fs/fs_code.cpp:44]: Seems like there should be a pattern for constructing into a storage and creating a guard for it all in one function.

File: libraries/libstratosphere/source/fs/fs_code.cpp [hunk 2]
```diff
@@ -20,6 +20,55 @@ namespace ams::fs {
...
+
+                /* Create the filesystem. */
+                std::construct_at(GetPointer(g_stratosphere_romfs_fs));
+                auto fs_guard = SCOPE_GUARD { std::destroy_at(GetPointer(g_stratosphere_romfs_fs)); };
```
Comment (misson20000) [inline libraries/libstratosphere/source/fs/fs_code.cpp:48]: Ditto

**Comment by SciresM** [comment]
Got a "seems fine" from fincs, and a smiley face from yellows8, so I'm merging this under the presumption that we're good.

**Comment by fennectech** [comment]
This doesnt exactly recreate the functionality requested here. It doesnt allow us to separate game mods from user installed sysmodules for easy backup of mods without backing up user sysmodules.   A separate contents folder tor romfs mods that separates from user modules would make it much easier for working with lots of mods.   Picking mods from modules is a pain. Even without atmosphere modules being in there.

**Pull request merged**